server/server.ts

```diff
@@ -13,6 +30,164 @@ const db = new pg.Pool({
+    const entry = result.rows;
+    if (!entry) {
+      throw new ClientError(404, 'No entries are available');
+    }
+    res.json(entry);
+  } catch (error) {
+    next(error);
+  }
+});
+
+app.get('/api/details/:entryId', authMiddleware, async (req, res, next) => {
+  try {
+    const { entryId } = req.params;
+    if (!entryId) throw new ClientError(400, 'please provide entryId.');
+    const sql = `select * from "entries"
+                where "entryId" = $1 and "userId" = $2;`;
+    const result = await db.query<Entry>(sql, [entryId, req.user?.userId]); // as Entry
+    const entry = result.rows[0];
+    if (!entry) {
+      throw new ClientError(404, 'No entries are available');
+    }
+    res.json(entry);
+  } catch (error) {
+    next(error);
+  }
+});
+
+app.post('/api/entry', authMiddleware, async (req, res, next) => {
+  try {
+    const { title, notes, photoUrl } = req.body;
+    if (!title) throw new ClientError(400, 'please provide title.');
+    if (!notes) throw new ClientError(400, 'please provide notes.');
+    if (!photoUrl) throw new ClientError(400, 'please provide photoUrl.');
+    const sql = `insert into "entries" ("title", "notes", "photoUrl", "userId")
+      values ($1,$2,$3, $4)
+      returning * `;
+    const params = [title, notes, photoUrl, req.user?.userId];
+    const result = await db.query<Entry>(sql, params); // as Entry
+    console.log(result);
+    const entry = result.rows[0];
+    if (!entry) {
+      throw new ClientError(404, 'No entries are available');
+    }
+    res.json(entry);
+  } catch (error) {
+    next(error);
+  }
+});
+
+app.put('/api/details/:entryId', authMiddleware, async (req, res, next) => {
+  try {
+    const { title, notes, photoUrl } = req.body;
+    const { entryId } = req.params;
+    if (!title) throw new ClientError(400, 'please provide title.');
+    if (!notes) throw new ClientError(400, 'please provide notes.');
+    if (!photoUrl) throw new ClientError(400, 'please provide photoUrl.');
+    if (!entryId) throw new ClientError(400, 'please provide entryId.');
+    const sql = `update "entries"
+    set "title" = $1, "notes" = $2, "photoUrl" = $3
+    where "entryId" = $4 and "userId" = $5
+    returning * ;`;
+    const params = [title, notes, photoUrl, entryId, req.user?.userId];
+    const result = await db.query<Entry>(sql, params); // as Entry
+    const entry = result.rows[0];
+    if (!entry) {
+      throw new ClientError(404, 'No entries are available');
+    }
+    res.json(entry);
+  } catch (error) {
+    next(error);
+  }
+});
+
+app.delete('/api/details/:entryId', authMiddleware, async (req, res, next) => {
+  try {
+    const { entryId } = req.params;
+    if (!entryId) throw new ClientError(400, 'please provide entryId.');
+    const sql = `delete from "entries"
+    where "entryId" = $1 and "userId" = $2
+    returning * ;`;
+    const result = await db.query<Entry>(sql, [entryId, req.user?.userId]);
+    console.log(result);
```

Comment:
This log message doesn't look like it's very useful anymore. Probably there for debugging, so should be removed now.